airframe-codec/.jvm/src/main/scala/wvlet/airframe/codec/Compat.scala

```diff
@@ -48,6 +48,20 @@ object Compat {
     JavaTimeCodec.javaTimeCodecs ++ JavaStandardCodec.javaStandardCodecs
 
   def codecOf[A: ru.TypeTag]: MessageCodec[A] = MessageCodecFactory.defaultFactory.of[A]
+  def codecOfClass(
+      cl: Class[_],
+      codecFactory: MessageCodecFactory = MessageCodecFactory.defaultFactoryForJSON
+  ): Option[MessageCodec[_]] = {
+    // Finding the surface using reflection
+    ReflectSurfaceFactory.ofClass(cl) match {
+      case g: GenericSurface if g.params == 0 =>
```

Comment:
I think this can work in most cases, it's okay. Just a thought, for instance, case object is not covered?

Reply:
case objects will be resolved as EnumSurface so this pass will not be used, but let me double-check that.

Reply:
And that was a good catch. It didn't work for case objects. I revised this logic to be specific to the case when ObjectCodec or ObjectMapCodec has empty parameters.